### DISCUSSION TITLE
Qiita を使用していると `Request Header Or Cookie Too Large` のエラーがでる

### DISCUSSION THREAD

**Discussion by yoshi389111** [increments/qiita-discussions#489] [要望・アイデア・不具合]
細かい発生条件はわからないのですが、Qiita を使っていると `400 Bad Request` でエラーになることがあります（２回発生しました）。
一度エラーになると、Qiita の全ページで `400 Bad Request` になってしまいます。

`400 Bad Request` のエラー画面には `Request Header Or Cookie Too Large` と記載がありました。

通常のログアウトもできないので、`qiita.com` のクッキーを削除してから再ログインすると、それ以降は正常に表示できるようになりました。

１回目は 2023-04-18 、２回目は 2023-06-13 に発生しました。

２回目の時に Chrome の開発者ツールでエラーになった Request ヘッダーの `Cookie` を見てみたところ、 `_ga_**********=GS1.1.NNNNNNNNNN.N.N.NNNNNNNNNN.N.0.0` のような形式のクッキーが 124 個登録されているようでした（`*`は英数字、`N` は数字）。
また、`Cookie` の値は、文字列として 8192 文字になっていました。

おそらく nginx の定義でリクエストヘッダーの行が約 8k になるとエラーとなる設定になっており、何らかの操作で `_ga_**********` というクッキーが増えていって、この制限を超えたところで `400 Bad Request` が起きるのではないかと思っています。

お手数ですがご確認をお願いします。

今回（２回目）の発生環境は Windows 11 (22H2) + Chrome  114.0.5735.110 です。
前回（１回目）の発生環境も同じですが、細かいバージョンはわかりません。

**Comment by tanaka-asami415** [comment]
@yoshi389111 
ご報告いただきありがとうございます。

ご不便をおかけして申し訳ございませんが、発生した際にはCookie削除で対応いただけますでしょうか🙏
調査をしてみますが、いただいた情報から推測すると解決が難しい可能性があります。ご了承いただけると幸いです🙇

**Comment by JunichiIto** [comment]
僕もさっき同じエラーに遭遇しました。
Cookieの中身はチェックしてないのですが、とりあえず対処方法をブログにまとめておきました。
取り急ぎご参考までに。

[Qiitaで"400 Bad Request"が表示されてアクセスできなくなった場合の対処方法 \- give IT a try](https://blog.jnito.com/entry/2023/07/14/103141)

**Reply by tanaka-asami415** [reply]
@JunichiIto 
Twitterと返信が重複してしまいますが、ありがとうございます！🙏
とても助かります！

**Comment by tanaka-asami415** [comment]
こちら調査を行ったところ、GoogleAnalyticsのcookieの保存期間・方法を変更することによって解決しそうなので、試してみます！
来週より着手予定ですので、もうしばらくお待ちいただけると幸いです🙇

**Reply by JunichiIto** [reply]
ご報告ありがとうございます！よろしくお願いします〜。

**Comment by tanaka-asami415** [comment]
@yoshi389111 @JunichiIto 
予定より早く着手ができたため、本日対応が完了いたしました！
ご報告いただいてからお待たせしてしまい申し訳ありません🙇

GoogleAnalyticsのcookieの保存期間を変更しました。
しばらく様子を見ていただき、再度現れるようであれば教えていただけると幸いです🙏

**Reply by JunichiIto** [reply]
ご対応ありがとうございます！了解しました〜。

**Comment by ko1nksm** [comment]
どうも完全には修正されていないようです。昨日同様の事例が発生してから `_ga_` で始まるクッキーのみを（ログアウトされてくないのでChromeのDevToolsから）削除したのですが、また新たにできています。タイミングなどよくわかりませんが一つずつ増えて現在４つになりました。ブラウザから見ると Expire/Max-Age が 400 日後ぐらいに設定されています。ただし一つだけ一週間ぐらいに設定されているものがあるので、対応できていないページがあるのではないかと推測されます。

**Reply by getty104** [reply]
ご報告ありがとうございます :pray: 
具体的にどのような状況か把握を行いたいので、以下の情報を[お問い合わせフォーム](https://support.qiita.com/hc/ja/requests/new)よりお送りいただけないでしょうか？

- 追加されているGAのCookie名
- それぞれのCookieに設定されている値
- それぞれのCookieの有効期限

お手数おかけしてしまい申し訳ございませんが、よろしくお願いします！

**Reply by ko1nksm** [reply]
返信いたしました。また増えていますね。昨日はそんなに増えなかったと思うのですが
ページを開いたまま放置していると増える？条件はよくわかりません。

**Reply by getty104** [reply]
早速お送りいただきありがとうございます！
いただいた内容をもとに、調査いたします :pray:

**Reply by ohakutsu** [reply]
おまたせしました！
こちら、有効期限の変更が一部漏れていた箇所の修正と再発しにくくなるように有効期限の設定を見直しました。
また、別途再発しにくくなるような対応を実施予定です！

**Reply by ko1nksm** [reply]
ご対応ありがとうございます！

**Comment by tanaka-asami415** [comment]
10月に別途再発しにくくなる修正を行い、一定期間様子を見ておりましたが
同様の問題は起こりにくくなったと判断し、本件クローズさせていただきます。

もし、また再発するようであれば教えていただけると幸いです。
ご報告とご協力をいただきありがとうございました🙇
